projects/extension/sql/idempotent/017-litellm.sql

```diff
@@ -0,0 +1,58 @@
+-------------------------------------------------------------------------------
+-- litellm_embed
+-- generate an embedding from a text value
+create or replace function ai.litellm_embed
+( model text
+, input_text text
+, api_key text default null
+, api_key_name text default null
+, extra_options jsonb default null
+) returns @extschema:vector@.vector
+as $python$
+    #ADD-PYTHON-LIB-DIR
+    import ai.litellm
+    import ai.secrets
+    options = {}
+    if extra_options is not None:
+        import json
+        options = {k: v for k, v in json.loads(extra_options).items()}
+
+    api_key_resolved = ai.secrets.get_secret(plpy, api_key, api_key_name, None, SD)
```

Comment:
Is an api_key always required? If so, we should abort before making the API call.

Reply:
No, it's not required.